src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt

```diff
@@ -149,13 +143,20 @@ class ProtectLevelCalculator(
       }
     }.also { log.debug("Additional Factors for Women Points $it") }
 
-  private fun hasArsonOrViolence(convictions: Collection<Conviction>): Boolean =
-    convictions.flatMap { it.offences }
+  private fun getArsonOrViolencePoints(convictions: Collection<Conviction>): Int =
+    if (convictions.flatMap { it.offences }
       .map { it.offenceDetail }.any {
         OFFENCE_CODES.contains(it.mainCategoryCode)
       }
+    ) 2 else 0
+
+  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int {
+    val sentences = convictions.map { it.sentence }
+    val longerThanTenMonths = sentences.any { it.startDate != null && it.expectedSentenceEndDate != null && Period.between(it.startDate, it.expectedSentenceEndDate).months >= 10 }
+    val indeterminateSentence = sentences.any { it.latestCourtAppearanceOutcome?.code == "303" }
```

Comment:
There is definitely no need to change this, but again here I feel that Sentence should either be suffixed to both vals or neither (neither is more readable in my view)